File: CommonRecycler/src/main/java/com/shuyu/bind/holder/BindErrorHolder.java
package com.shuyu.bind.holder;

import android.content.Context;
import android.text.Html;
import android.view.View;
import android.widget.TextView;

import com.shuyu.bind.BindRecyclerBaseHolder;
import com.shuyu.common.R;

/**
 * 数据为绑定等错误提示
 * Created by guoshuyu on 2017/8/29.
 */

public class BindErrorHolder extends BindRecyclerBaseHolder {

    public final static int ID = R.layout.error_item;

    private TextView errorText;

    public BindErrorHolder(View v) {
        super(v);
    }

    public BindErrorHolder(Context context, View v) {
        super(context, v);
    }

    @Override
    public void createView(View v) {
        errorText = (TextView) v.findViewById(R.id.error_text);
    }

    @Override
    public void onBind(Object model, int position) {
        String text = "model：<font color=\"#000000\"><big>" + model.getClass().getName() + "</big></font>    never bind";
        errorText.setText(Html.fromHtml(text));
    }
}

File: CommonRecycler-Kotlin/src/main/java/com/shuyu/commonrecycler/holder/BindErrorHolder.kt
package com.shuyu.commonrecycler.holder

import android.content.Context
import android.text.Html
import android.view.View
import android.widget.TextView

import com.shuyu.commonrecycler.BindRecyclerBaseHolder
import com.shuyu.commonrecycler.R

/**
 * 数据为绑定等错误提示
 * Created by guoshuyu on 2017/8/29.
 */

open class BindErrorHolder : BindRecyclerBaseHolder {

    private var errorText: TextView? = null

    constructor(v: View) : super(v) {}

    constructor(context: Context, v: View) : super(context, v) {}

    override fun createView(v: View) {
        errorText = v.findViewById(R.id.error_text) as TextView
    }

    override fun onBind(model: Any, position: Int) {
        val text = "model：<font color=\"#000000\"><big>" + model.javaClass.name + "</big></font>    never bind"
        errorText!!.text = Html.fromHtml(text)
    }

    companion object {

        val ID = R.layout.error_item
    }
}
